extensions/wikia/SpecialDiscussions/SpecialDiscussionsController.class.php

```diff
@@ -41,9 +34,13 @@ public function index() {
 
 		$this->wg->Out->setPageTitle( wfMessage( 'discussions-pagetitle' )->escaped() );
 
-		if ( $this->request->wasPosted() ) {
-			$this->activateDiscussions();
-			WikiFactory::setVarByName( 'wgEnableDiscussions', $this->siteId, true );
+		if ( $this->request->wasPosted() && $this->assertValidPostRequest() ) {
+			if ( !SpecialDiscussionsHelper::activateDiscussions(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1066)